nav2_rviz_plugins/src/costmap_cost_tool.cpp

```diff
@@ -0,0 +1,137 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_rviz_plugins/costmap_cost_tool.hpp"
+#include <rviz_common/viewport_mouse_event.hpp>
+#include "rviz_common/display_context.hpp"
+#include "rviz_common/load_resource.hpp"
+#include "rviz_common/interaction/view_picker_iface.hpp"
+#include "rviz_common/msg_conversions.hpp"
+#include "rviz_common/properties/bool_property.hpp"
+
+namespace nav2_rviz_plugins
+{
+CostmapCostTool::CostmapCostTool()
+: qos_profile_(5)
+{
+  shortcut_key_ = 'm';
+
+  auto_deactivate_property_ = new rviz_common::properties::BoolProperty(
+    "Single click", true,
+    "Switch away from this tool after one click.",
+    getPropertyContainer(), SLOT(updateAutoDeactivate()), this);
+}
+
+CostmapCostTool::~CostmapCostTool() {}
+
+void CostmapCostTool::onInitialize()
+{
+  hit_cursor_ = cursor_;
+  std_cursor_ = rviz_common::getDefaultCursor();
+
+  setName("Costmap Cost");
+  setIcon(rviz_common::loadPixmap("package://rviz_default_plugins/icons/classes/PointStamped.png"));
+
+  node_ = context_->getRosNodeAbstraction().lock()->get_raw_node();
+  local_cost_client_ =
+    node_->create_client<nav2_msgs::srv::GetCost>("/local_costmap/get_cost_local_costmap");
+  global_cost_client_ =
+    node_->create_client<nav2_msgs::srv::GetCost>("/global_costmap/get_cost_global_costmap");
+}
+
+void CostmapCostTool::activate() {}
+void CostmapCostTool::deactivate() {}
+
+int CostmapCostTool::processMouseEvent(rviz_common::ViewportMouseEvent & event)
+{
+  int flags = 0;
+
+  Ogre::Vector3 position;
+  bool success = context_->getViewPicker()->get3DPoint(event.panel, event.x, event.y, position);
+  setCursor(success ? hit_cursor_ : std_cursor_);
+
+  if (success) {
+    std::ostringstream s;
+    s << "<b>Left-Click:</b> Select this point.";
+    s.precision(3);
+    s << " [" << position.x << "," << position.y << "," << position.z << "]";
+    setStatus(s.str().c_str());
+
+    if (event.leftUp()) {
+      auto point = rviz_common::pointOgreToMsg(position);
+
+      callCostService(point.x, point.y);
+
+
+      if (auto_deactivate_property_->getBool()) {
+        flags |= Finished;
+      }
+    }
+  } else {
+    setStatus("Move over an object to select the target point.");
+  }
+  return flags;
+}
+
+void CostmapCostTool::callCostService(float x, float y)
+{
+  // Create request for local costmap
+  auto request = std::make_shared<nav2_msgs::srv::GetCost::Request>();
+  request->x = x;
+  request->y = y;
+
+  // Call local costmap service
+  if (!local_cost_client_->wait_for_service(std::chrono::seconds(1))) {
```

Comment:
Change this logic to: If service is available, then call it to check the cost and print the cost in the rviz terminal

Don't return when local / global aren't available. Some applications may only have one or the other.